File: C++/source/Framework/Messaging/Local/WaveServiceIndependentMessage.cpp
/***************************************************************************
 *   Copyright (C) 2012 Vidyasagara Guntaka                                
 *   All rights reserved.  
 *   Description: This file implements the service independent message
 *                class.  Service independent messages allow for immediate 
 *                handling of the message across any transport (Client,
 *                HaPeer, InterLocation).
 *   Author : Brian Adaniya 
 *   Date   : 10/18/2012                                           
 ***************************************************************************/

#include "Framework/Messaging/Local/WaveServiceIndependentMessage.h"

namespace WaveNs
{

WaveServiceIndependentMessage::WaveServiceIndependentMessage (WaveServiceId serviceCode, UI32 operationCode)
    : WaveMessage (serviceCode, operationCode),
    WaveServiceIndependentMessageBase ()
{
}

WaveServiceIndependentMessage::~WaveServiceIndependentMessage ()
{
}

void WaveServiceIndependentMessage::setupAttributesForSerialization ()
{
    WaveMessage::setupAttributesForSerialization (); 
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/Local/WaveServiceIndependentMessage.java
/***********************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 ***********************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.Local;

import com.CxWave.Wave.Framework.Type.WaveOperationCodeInterface;
import com.CxWave.Wave.Framework.Type.WaveServiceId;

public class WaveServiceIndependentMessage extends WaveMessage implements WaveServiceIndependentMessageBase
{

    protected WaveServiceIndependentMessage (final WaveServiceId serviceCode, final WaveOperationCodeInterface operationCode)
    {
        super (serviceCode, operationCode);
    }

}
